Handle "binary-only" releases

Many Release Candidates and virtually every Nightly node release doesn't include every single platform. The way the current binary support works, anything missing just falls back to a from-source compile. However, there are also rare instances where a nightly or release candidate doesn't include a _source_ package.

I'm not exactly sure how the 'nightly' definition handles this right now (I believe it just skips it entirely, falling back to a previous nightly). However, now that the node-build definition scraper supports published Release Candidates; we should probably handle these rare instances where a build definition is _only_ binaries.

Turns out, having `install_package "foo"` is sufficient to allow the binaries to install.

```
binary linux-x64 "https://nodejs.org/dist/v11.4.0/node-v11.4.0-linux-x64.tar.gz#4ef7dea131453da3a93cc1f32bff948da8953958dcbe3b413debae8bb41aa7a0"

install_package "node-v11.4.0"
```

In the example above, the binary will install correctly on linux-x64 and no error message is generated for the missing source package.

However, any other platform will get a failure when it attempts to download an empty url:

```
Downloading ...

BUILD FAILED (OS X 10.13.6 using node-build 3.1.0-14-gee4838a)
```

So the only improvement to make here, is to make the downloader exit with a more helpful message when the URL is empty.

--- a/bin/node-build
+++ b/bin/node-build
@@ -461,6 +461,11 @@ fetch_tarball() {
   local checksum
   local extracted_dir
 
+  if [ -z "$package_url" ]; then
+    echo "error: failed to download $package_name (missing package url)" >&2
+    return 1
+  fi
+
   if [ "$package_url" != "${package_url/\#}" ]; then
     checksum="${package_url#*#}"
     package_url="${package_url%%#*}"

